test/client/audio_consumer_test.py

```diff
@@ -1,171 +1,144 @@
             monitor['pos_end'] = message.get('pos_end')
 
         self.loop.once('recognizer_loop:wakeword', wakeword_callback)
-        self.consumer.try_consume_audio()
+        self.consumer.read_audio()
 
-        pos_begin = monitor.get('pos_begin')
-        self.assertIsNotNone(pos_begin)
-        diff = abs(pos_begin - TRUE_POS_BEGIN)
+        actual_begin = monitor.get('pos_begin')
+        self.assertIsNotNone(actual_begin)
+        diff = abs(actual_begin - ideal_begin)
         self.assertTrue(
-            diff <= TOLERANCE_RANGE_FRAMES,
-            str(diff) + " is not less than " + str(TOLERANCE_RANGE_FRAMES))
+                diff <= tolerance,
+                str(diff) + " is not less than " + str(tolerance))
 
-        pos_end = monitor.get('pos_end')
-        self.assertIsNotNone(pos_end)
-        diff = abs(pos_end - TRUE_POS_END)
+        actual_end = monitor.get('pos_end')
+        self.assertIsNotNone(actual_end)
+        diff = abs(actual_end - ideal_end)
         self.assertTrue(
-            diff <= TOLERANCE_RANGE_FRAMES,
-            str(diff) + " is not less than " + str(TOLERANCE_RANGE_FRAMES))
+                diff <= tolerance,
+                str(diff) + " is not less than " + str(tolerance))
 
     def test_wakeword_in_beginning(self):
-        self.queue.put(self.__create_sample_from_test_file('mycroft'))
+        self.queue.put(self.__create_sample_from_test_file('weather_mycroft'))
+        self.recognizer.set_transcriptions(["what's the weather next week"])
         monitor = {}
-        self.recognizer.set_transcriptions([
-            "what's the weather next week", ""])
 
         def callback(message):
             monitor['utterances'] = message.get('utterances')
 
         self.loop.once('recognizer_loop:utterance', callback)
-        self.consumer.try_consume_audio()
+        self.consumer.read_audio()
+
         utterances = monitor.get('utterances')
         self.assertIsNotNone(utterances)
         self.assertTrue(len(utterances) == 1)
         self.assertEquals("what's the weather next week", utterances[0])
 
-    def test_wakeword_in_phrase(self):
+    def test_wakeword(self):
         self.queue.put(self.__create_sample_from_test_file('mycroft'))
+        self.recognizer.set_transcriptions(["silence"])
         monitor = {}
-        self.recognizer.set_transcriptions([
-            "he can do other stuff too", "what's the weather in cincinnati"])
 
         def callback(message):
             monitor['utterances'] = message.get('utterances')
 
         self.loop.once('recognizer_loop:utterance', callback)
-        self.consumer.try_consume_audio()
-        utterances = monitor.get('utterances')
-        self.assertIsNotNone(utterances)
-        self.assertTrue(len(utterances) == 2)
-        self.assertEquals("he can do other stuff too", utterances[0])
-        self.assertEquals("what's the weather in cincinnati", utterances[1])
-
-    def test_call_and_response(self):
```

Comment:
I do not see the behavior of this test captured anywhere. This test verifies call and response behavior. The expected transcriptions has changed a bit, but the behavior of the state machine transitioning from "waiting for wakeword" to "record audio regardless of wakeword" needs to be verified via test. It has broken several times in the past.